improver_tests/set_up_test_cubes.py

```diff
@@ -111,11 +109,11 @@ def _create_yx_arrays(ypoints, xpoints, domain_corner, grid_spacing):
         Tuple[numpy.ndarray, numpy.ndarray]:
             Tuple containing arrays of y and x coordinate values
     """
-    y_stop = domain_corner[0] + (grid_spacing * ypoints)
-    x_stop = domain_corner[1] + (grid_spacing * xpoints)
+    y_stop = domain_corner[0] + (grid_spacing * (ypoints - 1))
+    x_stop = domain_corner[1] + (grid_spacing * (xpoints - 1))
 
-    y_array = np.arange(domain_corner[0], y_stop, grid_spacing, dtype=np.float32)
-    x_array = np.arange(domain_corner[1], x_stop, grid_spacing, dtype=np.float32)
+    y_array = np.linspace(domain_corner[0], y_stop, ypoints, dtype=np.float32)
+    x_array = np.linspace(domain_corner[1], x_stop, xpoints, dtype=np.float32)
```

Comment:
What was the reason for changing this?  I preferred `arange` as it enforces the grid spacing directly in a way that's difficult to break accidentally in the future.

Reply:
I found with both 'arange' and 'linspace', when floating point numbers were used they can become unreliable at returning evenly spaced values, when I looked again at the documentation I noticed that they [advise to use linspace when non-integer steps are used](https://numpy.org/doc/stable/reference/generated/numpy.arange.html). I was thinking that non-integer steps may be common for lat/lon grids?

If you change it back to use `arange`, 9 of the tests in `test_RegridLandSea.py` fail. I think this is because the start and stop values are floats, and the documentation states that:

**stop : _number_**
    End of interval. The interval does not include this value, except in some cases where step is not an integer and floating point round-off affects the length of out.

Whereas with `linspace` unless explicity told not to, the stop number will be included in the array.

Reply:
OK, that sounds sensible. Thanks for explaining.